Its.Log.Monitoring/StatefulResult.cs

```diff
@@ -0,0 +1,14 @@
+namespace Its.Log.Monitoring
+{
+    public class StatefulResult<TState, TResult>
+    {
+        public StatefulResult(TState state, TResult result)
+        {
+            State = state;
+            Result = result;
+        }
+
+        public TResult Result { get; }
+        public TState State { get; }
+    }
+}
```

Comment:
I think that `Result` is just as informative as `StatefulResult`. The word "state" (including as a property name) is very general. I wonder if we can come up with something more specific.